onnxruntime/core/framework/prepacked_weights_cache.h

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/framework/buffer_deleter.h"
+#include "core/framework/tensor_shape.h"
+#include "core/framework/allocator.h"
+#include <unordered_map>
+#include <string>
+
+namespace onnxruntime {
+
+struct PrepackedWeight {
+  // Some weights may be associated with multiple pre-packed buffers.
+  // Hence we hold them in containers. It is upto the developer implementing each PrePack()
+  // method to define what gets stored in which position of the containers.
+
+  // NOTE: Not all fields may be filled in and not all containers will have the same number of elements
+  // It is upto the developer of the kernel to decide which fields to cache for re-use.
+
+  std::vector<std::unique_ptr<void, BufferDeleter>> buffers_;  // cache pre-packed buffers associated with the kernel
+  std::vector<size_t> weights_sizes_;                          // cache sizes associated with pre-packed buffers
+  std::vector<TensorShape> shapes_;                            // cache tensor shapes associates with pre-packed buffers
+  std::vector<bool> flags_;                                    // cache some flags associated with the pre-packed buffers
+
+  bool has_cached_ = false;
```

Comment:
Any reason this is still required? Why should an object know whether it's cached or not?

Reply:
Changed this to `is_filled_`